### PR TITLE
chore: avoid no-ack when ctx is canceled

### PR DESCRIPTION
**Pull request by yhl25**
* fixes flaky reduce-e2e tests

### CODE REVIEW

File: pkg/reduce/pbq/pbq.go [hunk 1]
```diff
@@ -68,11 +68,13 @@ func (p *PBQ) Write(ctx context.Context, request *window.TimedWindowRequest, per
 	}
 
 	// write the request to the output channel
+	// since it is a blocking write, we should have a select with context,
 	select {
 	case p.output <- request:
-
 	case <-ctx.Done():
-		return ctx.Err()
+		// we can persist the message even if the context is done that way we will not rely on
```
Comment (vigith) [inline pkg/reduce/pbq/pbq.go:75]: How will we guarantee that the message will be written to disk?

Comment (yhl25) [reply pkg/reduce/pbq/pbq.go:75]: Writing to disk is a blocking call, and we only close the WAL once we're done processing all the messages in the batch.

Comment (yhl25) [reply pkg/reduce/pbq/pbq.go:75]: Just like No-Ack, WAL writes also don't depend on the ctx.

Comment (vigith) [reply pkg/reduce/pbq/pbq.go:75]: Please open an issue to track

**Review by vigith — APPROVED** [review]

**Review by whynowy — APPROVED** [review]

**Pull request merged**